components/common/ActionButton.tsx

```diff
@@ -1,29 +1,34 @@
 'use client';
+
 import { Button } from '@/components/ui/button';
 import { useState, ReactNode } from 'react';
 
 type Props = {
-  onClick: () => void;
+  onClick?: () => void;
   icon?: ReactNode;
   children: ReactNode;
+  type?: 'button' | 'submit' | 'reset';
+  className?: string;
 };
 
-export const ActionButton = ({ icon, onClick, children }: Props) => {
+export const ActionButton = ({ icon, onClick, children, type = 'button', className }: Props) => {
   const [isCompleted, setIsCompleted] = useState(false);
 
   const handleClick = () => {
     setIsCompleted(!isCompleted);
-    onClick();
+    onClick?.();
   };
 
   return (
     <Button
+      type={type}
       onClick={handleClick}
       variant="ghost"
       className={`
         bg-transparent hover:bg-transparent shadow-none border-none px-0 py-0 h-auto
         flex items-center gap-1 text-sm font-bold text-secondary
         hover:text-primary transition-colors duration-200
+        ${className ?? ''}
```

Comment:
@Yoonsoyoung02 
여기서 이렇게 처리하지 말구 위에 props 에서 default 값을 지정해주시죠!
```suggestion
    ({ icon, onClick, children, type = 'button', className = ''}: Props)
```

Reply:
넵 수정 완료했습니다~!